core/src/com/unciv/logic/city/managers/CityEspionageManager.kt

```diff
@@ -20,4 +23,22 @@ class CityEspionageManager : IsPartOfGameInfoSerialization{
         return civInfo.espionageManager.spyList.any { it.location == city.id }
     }
 
+    private fun getAllStationedSpies(): List<Spy> {
+        return city.civ.gameInfo.civilizations.flatMap { civ ->
+            civ.espionageManager.spyList.filter { it.location == city.id }
+        }
+    }
+
+    fun removeAllPresentSpies(reason: String) {
```

Comment:
Please make this an enum, not hardcoded strings

Reply:
Will do